src/en/clowncorps/src/eu/kanade/tachiyomi/extension/en/clowncorps/ClownCorps.kt

```diff
@@ -0,0 +1,241 @@
+import kotlinx.serialization.json.Json
+import okhttp3.Response
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ClownCorps : ConfigurableSource, HttpSource() {
+    override val baseUrl = "https://clowncorps.net"
+    override val lang = "en"
+    override val name = "Clown Corps"
+    override val supportsLatest = false
+
+    override val client = network.client.newBuilder()
+        .addInterceptor(TextInterceptor())
+        .build()
+
+    private fun getManga() = SManga.create().apply {
+        title = name
+        artist = CREATOR
+        author = CREATOR
+        status = SManga.ONGOING
+        initialized = true
+        // Image and description from: https://clowncorps.net/about/
+        thumbnail_url = "$baseUrl/wp-content/uploads/2022/11/clowns41.jpg"
+        description = "Clown Corps is a comic about crime-fighting clowns.\n" +
+            "It's pronounced \"core.\" Like marine corps."
+        url = "/comic"
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> =
+        Observable.just(MangasPage(listOf(getManga()), hasNextPage = false))
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList) =
+        fetchPopularManga(page)
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        Observable.just(getManga())
+
+    @Serializable
+    data class SerializableChapter(val fullLink: String, val name: String, val dateUpload: Long) {
+        override fun hashCode(): Int {
+            return name.hashCode()
+        }
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as SerializableChapter
+
+            return name == other.name
+        }
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        // The total number of webpages with chapters on them
+        val currentPageIndicator = response.asJsoup().select("#paginav li.paginav-pages").text()
+        val totalWebpageCount = currentPageIndicator.split(" ").last().toInt()
+
+        val allChapters = getChaptersFromCache().toMutableSet()
+        // Fetch all the chapters from the website until we reached where the cache left off
+        for (webpageIndex in 1..totalWebpageCount) {
+            val anyChaptersWereAdded = allChapters.addAll(fetchWebpageChapters(webpageIndex))
+            if (!anyChaptersWereAdded) break // No new chapters were added from this webpage, so we're done
+        }
+
+        // Save the chapters to cache
+        val fullJsonString = Json.encodeToString(allChapters)
+        preferences.edit().putString(CACHE_KEY_CHAPTERS, fullJsonString).apply()
+
+        // Convert the serializable chapters to SChapters
+        return allChapters.map { chapter ->
```

Comment:
How are you maintaining the chapters' order? It should be sorted from new to old. Either sort it by upload date (and hope post edits don't reset it) or maintain the order from the beginning.

Reply:
Done in ac05b17e338f4d0a5ebccce441b9b4fbba025135